src/ngMessages/messages.js

```diff
@@ -102,6 +125,60 @@
  * email addresses, date fields, autocomplete inputs, etc...), specialized error messages can be applied
  * while more generic messages can be used to handle other, more general input errors.
  *
+ * ## Dynamic Messaging
+ * In AngularJS 1.4 ngMessages supports using expressions to dynamically change key values. Using arrays
+ * and repeaters to list messages is also supported. This means that the code below will be able to fully
+ * adapt itself and display the appropriate message when any of the expression data changes:
+ *
+ * ```html
+ * <form name="myForm">
+ *   <input type="email"
+ *          name="myEmail"
+ *          ng-model="email"
+ *          minlength="5"
+ *          required />
+ *
+ *   <div ng-messages="myForm.myEmail.$error">
+ *     <div ng-message="required">You did not enter your email address</div>
+ *     <div ng-repeat="errorMessage in errorMessages">
+ *       <!-- use ng-message-exp for a message that is an expression -->
+ *       <div ng-message-exp="errorMessage.type">{{ errorMessage.text }}</div>
+ *     </div>
+ *   </div>
+ * </form>
+ * ```
+ *
+ * The `errorMessage.type` variable can be a string value or it can be an array that can have
+ * multiple error states be associated with a single error message:
+ *
+ * ```html
+ *  <input type="email"
+ *         ng-model="data.email"
+ *         name="myEmail"
+ *         ng-minlength="5"
+ *         ng-maxlength="100"
+ *         required />
+ *   <div ng-messages="myForm.myEmail.$error">
+ *     <div ng-message-exp="'required'">You did not enter your email address</div>
+ *     <div ng-message-exp="['minlength', 'maxlength]">
+ *       Your email must be between 5 and 100 characters long
+ *     </div>
+ *   </div>
+ * ```
+ *
+ * Feel free to use other directives such as ng-if and ng-switch to further control what messages are active
+ * and when. This, however, may cause some issues with Angular not being able to determine if a directive
+ * is active or not. Therefore if an error does occur it's best to place the ng-message element on its own
+ * element within the structural directive HTML code:
```

Comment:
This paragraph makes sense when you see the example but I think it could be clearer in itself.
Perhaps something like:

> Feel free to use other structural directives such as ng-if and ng-switch to further control what messages are active and when. Be careful, if you place ng-message on the same element as these structural directives, Angular may not be able to determine if a message is active or not. Therefore it is best to place the ng-message on a child element of the structural directive.